receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -151,3 +235,91 @@ func (m *Metadata) getPodVolume(podUID string, volumeName string) (v1.Volume, er
 
 	return v1.Volume{}, fmt.Errorf("pod %q with volume %q not found in the fetched metadata", podUID, volumeName)
 }
+
+func (m *Metadata) getPodCPULimit(uid string) *float64 {
```

Comment:
Why do we need to return a pointer?

Reply:
I opted to return a pointer so that I could return `nil` instead of `0` for the unfound values since the stats summary uses pointers as well.  In the accumulator we'll be able to check if the value is `nil` and if it is we'll skip emitting the metric - this follows the pattern we use today for the other metrics we emit.

Reply:
Ok, but in `podResource.cpuLimit`, you use 0 to identify missing value. I think removing conversion to pointer will simplify a few functions

Reply:
Updated to return 0.